spring-aop/src/main/java/org/springframework/aop/framework/ProxyFactoryBean.java

```diff
@@ -522,8 +522,13 @@ private void addGlobalAdvisor(ListableBeanFactory beanFactory, String prefix) {
 				BeanFactoryUtils.beanNamesForTypeIncludingAncestors(beanFactory, Advisor.class);
 		String[] globalInterceptorNames =
 				BeanFactoryUtils.beanNamesForTypeIncludingAncestors(beanFactory, Interceptor.class);
-		List<Object> beans = new ArrayList<>(globalAdvisorNames.length + globalInterceptorNames.length);
-		Map<Object, String> names = new HashMap<>(beans.size());
```

Comment:
Beans.size() is equal to 0, because there is no element in beans at this time, even if the initial capacity is set for it.